src/scaler/daemonscaler/provider.rs

```diff
@@ -209,10 +210,20 @@ impl<S: ReadStore + Send + Sync + Clone> Scaler for ProviderDaemonScaler<S> {
 impl<S: ReadStore + Send + Sync> ProviderDaemonScaler<S> {
     /// Construct a new ProviderDaemonScaler with specified configuration values
     pub fn new(store: S, config: ProviderSpreadConfig, component_name: &str) -> Self {
-        let id = format!(
-            "{PROVIDER_DAEMON_SCALER_TYPE}-{}-{component_name}-{}-{}",
-            config.model_name, config.provider_reference, config.provider_link_name,
-        );
+        let id = if let Some(provider_config) = &config.provider_config {
```

Comment:
So thinking back on this, can you walk me though why we want to hash the provider config here? Within wasmcloud, the link name uniquely identifies the provider. In other words, there isn't a world where a provider with link name `foo` and config `{"blah": 2}` can exist at the same time as the same provider with link name `foo` and config `{"bar": "baz"}`.

Reply:
good question. 

so, assuming that `provider_config` does impact how the scaler/wadm handles a provider on the lattice, [my understanding](https://github.com/wasmCloud/wadm/issues/159) is that we want  a manifest change  `foo w/ {"blah": 2} => foo w/ {"blah": "baz" }` to trigger some steps that the scaler/wadm needs to take for a desired state. Basically, i was under the impression that, as things currently stand, `foo w/ {"blah": 2} => foo w/ {"blah": "baz" }`  doesn't instigate any state transition despite because a change of `provider_config: {"blah": 2}  =>  {"blah": "baz" }` is not reflected in the current  change detection implementation.

@brooksmtownsend  let me know if I misunderstood the intent of #159 with regards to `provider_config`